#160056174 Completes Continuous Integration set up on repository

chukwuemekachm opened the pull request:
#### What does this PR do?
- Set up continuous integration on the repository
#### Description of Task to be completed?
- Write tests for all components
- Add `.travis` file to configure tests on travis-ci
- Add .circle-ci/config.yml` to configure tests on circle-ci
- Setup repository on `travis-ci`, `coveralls`, `code-climate` and `circle-ci`
#### How should this be manually tested?
- Clone the repository
- Start up command line
- Navigate into the projects directory using `cd`
- Run 'npm install`
- Run `npm test`
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#160056174](https://www.pivotaltracker.com/story/show/160056174)
#### Screenshots (if appropriate)

<img width="1440" alt="screen shot 2018-09-03 at 2 20 33 pm" src="https://user-images.githubusercontent.com/33798252/44990549-3c309f80-af89-11e8-9f7a-9ae3e6304358.png">

<img width="1440" alt="screen shot 2018-09-03 at 2 20 42 pm" src="https://user-images.githubusercontent.com/33798252/44990550-3c309f80-af89-11e8-8448-942eb8df91c0.png">

#### Questions:
- None

Commit:
chore (setup): setup empty react app

- create an empty react app using create-react-app
- configure eslint
[finishes #160056172]

Commit:
removed router definition

chukwuemekachm committed:
chore(travis): Completes setup of travis

- Install and configure enzyme dependencies
- Add jest configurations to package.json
- Add .travis.yml file

[Completes #160056174]

chukwuemekachm committed:
chore(circleci): Completes circleci setup

- Add .circleci directory
- Add config.yml with circleci configs

chukwuemekachm committed:
chore(circle-ci): Resolves circle-ci integration

- Removes code-climate test reporter

[Completes #160056174]

yomigeek reviewed (commented):
This is fine, however, I think the hound issues need to be fixed. Good job!

chukwuemekachm committed:
chore(hound): Configure Hound-CI

- Add hound.yml config file to repo root dir

[Completes #160248956]

chukwuemekachm committed:
chore(tests): Rebase on staging

- Rebase current work on staging

Veraclins commented on .circleci/config.yml:
```diff
@@ -0,0 +1,20 @@
+version: 2
```
@chukwuemekachm good job here. I am wondering though, is there a reason why the config for circleci is in a directory instead of on the root of the repo like the others? If it is not a requirement of the integration, I believe having it on the root will make it consistent with the rest.

chukwuemekachm replied:
Nice observation @Veraclins. However, circle-ci version 2 requires a `.circle-ci` folder which contains lots of settings and config files.

chukwuemekachm replied:
Yes, Travis-ci and circle-ci basically do the same thing. I decided to add circle-ci due to the issues we had during our first phase with Travis-ci taking too long to build because it had to queue the build process thereby holding of merges until the build status has been verified. Travis-ci is the default test builder and coverage report submitter, but when it takes too long to build a branch we can fall back to the results from circle-ci and easily make decisions from that.

Veraclins replied:
I still think we can use one of them. If Travis is not reliable, we can forget it and use circle-ci. Unless there are some things that circle-ci cannot do.

Veraclins commented on src/tests/__snapshots__/App.test.js.snap:
```diff
@@ -0,0 +1,31 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
I believe this file was autogenerated. Like @tonymontaro said yesterday, committing autogenerated files could cause unnecessary merge conflicts. You may want to add this file to .gitignore.

chukwuemekachm replied:
Yes, this file was auto-generated. The idea for snapshot testing is to generate a snapshot, save it and then compare it with the react component generated by rendering a component in the test. This would ensure that the UI state does not change unexpectedly by actions from a team member. The convention is to make it human readable and the snapshot artifact should be committed alongside code changes and reviewed as part of our code review process.
I recommend the officaial jest documentation on snapshot testing. [here](https://jestjs.io/docs/en/snapshot-testing)
I will see that the badges are added ASAP.

Thanks.

Veraclins reviewed (changes requested):
@chukwuemekachm as already stated, this is a nice job. Just a quick question, is there a reason we are using both Travis-ci and Circle-ci? I understand they basically do the same thing.
Finally, are we having a separate task for adding the badges to the README file? I believe you can easily add the badges to the README to complete the task. Once again nice job bro!

chukwuemekachm committed:
chore(tests): Add CI badges to README.md

- Obtain all CI badges
- Add CI badges to README.md

Oluwafayokemi committed:
chore(style/readme): Set up sass pre-processing, Bootstrap 4.0 framework and update readme (#4)

- install sass-loader and node-sass
- Add bootstrap 4.0 cdn to index.html
- Write descriptive readme
[Finish #160056177]

chukwuemekachm committed:
chore(test): Rebase on develop

- Rebase branch on current state of develop

chukwuemekachm committed:
Merge branch 'staging' into chore/160056174/setup-tests

Veraclins reviewed (commented):
@chukwuemekachm, nice job here, my only reservation here is on the use of Travis and circle-ci. To me, this feels like asking two drivers to drive a single car. Sooner or later, there could be contradictions. However, if the team agrees on keeping both, I will have no objection. On the whole, good job bro!

Oluwafayokemi reviewed (approved):
Nice job Chima!

madeofhuman reviewed (approved):
Good job, man. Clinton has addressed most of the concerns. I agree with him though, that having Travis and CircleCI seems like an overkill. I feel we should stick with CircleCI.
